sa4u_z3/tu.py

```diff
@@ -64,22 +64,33 @@ def _mp_parse_tu(compile_command: cindex.CompileCommand, q: multiprocessing.Queu
         q.put(err)
 
 
-def _parse_tu(compile_command: cindex.CompileCommand, cache_path: Optional[str] = None) -> Optional[Union[cindex.TranslationUnit, SerializedTU]]:
-    '''Parses the translation unit.'''
-    if cache_path:
-        full_path = os.path.join(
-            compile_command.directory,
-            compile_command.filename,
-        )
-        serialized_tu = read_tu(
-            cache_path,
-            full_path,
-        )
-        modified_time = os.path.getmtime(full_path)
-        if serialized_tu.serialization_time >= modified_time:
-            logger.info(f'Using cached analysis for {full_path}')
-            return serialized_tu
+def get_stored_stu(compile_command: cindex.CompileCommand, cache_path: Optional[str]) -> Optional[SerializedTU]:
+    full_path = os.path.join(
+        compile_command.directory,
+        compile_command.filename,
+    )
+    in_memory_name = _translation_unit_file_path_to_filename(full_path)
```

Comment:
Maybe instead of `in_memory_name` we should call this something like `cache_key`?